daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -450,47 +443,42 @@ func (l *logStream) collectBatch() {
 // bytes per event (defined in maximumBytesPerEvent).  There is a fixed per-event
 // byte overhead (defined in perEventBytes) which is accounted for in split- and
 // batch-calculations.
-func (l *logStream) processEvent(events []wrappedEvent, unprocessedLine []byte, timestamp int64) []wrappedEvent {
-	bytes := 0
+func (l *logStream) processEvent(batch *eventBatch, unprocessedLine []byte, timestamp int64) {
 	for len(unprocessedLine) > 0 {
 		// Split line length so it does not exceed the maximum
 		lineBytes := len(unprocessedLine)
 		if lineBytes > maximumBytesPerEvent {
 			lineBytes = maximumBytesPerEvent
 		}
 		line := unprocessedLine[:lineBytes]
-		unprocessedLine = unprocessedLine[lineBytes:]
-		if (len(events) >= maximumLogEventsPerPut) || (bytes+lineBytes+perEventBytes > maximumBytesPerPut) {
-			// Publish an existing batch if it's already over the maximum number of events or if adding this
-			// event would push it over the maximum number of total bytes.
-			l.publishBatch(events)
-			events = events[:0]
-			bytes = 0
-		}
-		events = append(events, wrappedEvent{
+
+		event := wrappedEvent{
 			inputLogEvent: &cloudwatchlogs.InputLogEvent{
 				Message:   aws.String(string(line)),
 				Timestamp: aws.Int64(timestamp),
 			},
-			insertOrder: len(events),
-		})
-		bytes += (lineBytes + perEventBytes)
+			insertOrder: batch.count(),
+		}
+
+		switch batch.add(event, lineBytes) {
+		case errBatchCountTooLarge, errBatchSizeTooLarge:
```

Comment:
Both of these errors result in the exact same code being run.  Why not just return `bool` so you can do

```go
if ok := batch.add(event, lineBytes); ok {
	unprocessedLine = unprocessedLine[lineBytes:]
} else {
	l.publishBatch(batch)
}
```

Reply:
There's certainly simplicity in the `bool` approach, but I think that having the result codified is easier to reason about. I prefer to know that the there's a reason behind not adding the event, not just that it was added or not and the overhead to do so is negligible.

Reply:
You're not recording that anywhere though.  When I was reading the code, the different errors confused me because I didn't know how they were handled differently.  Since they're in fact handled the same way and we don't ever do anything with the reason it seems like an unnecessary distinction.

Reply:
You're right, either way we'll be handling it and arriving at the same result. This is a YAGNI situation, I'll drop this.